File: how-to-verify-images/python_examples/ImageTesting.py
import unittest
from PIL import Image


class ImageTesting(unittest.TestCase):

    def test_image_by_data_value(self):
        actual_image = Image.open('../images/original.png')
        expected_image = Image.open('../images/original_copy.png')
        self.assertEqual(
            list(actual_image.getdata()),
            list(expected_image.getdata())
        )

    def test_image_by_data_value_negative(self):
        actual_image = Image.open('../images/original.png')
        expected_image = Image.open('../images/edit.png')
        self.assertEqual(
            list(actual_image.getdata()),
            list(expected_image.getdata())
        )

File: how-to-verify-images/kotlin_examples/src/test/java/ImageTesting.kt
import org.testng.Assert.assertEquals
import org.testng.annotations.Test
import java.nio.file.Files
import java.nio.file.Path


class ImageTesting {

    @Test
    fun testImageByDataValuePositive() {
        val actualImage = Files.readAllBytes(Path.of("images/original.png")).toList()
        val expectedImage = Files.readAllBytes(Path.of("images/original_copy.png")).toList()
        assertEquals(actualImage, expectedImage)
    }

    @Test
    fun testImageByDataValueNegative() {
        val actualImage = Files.readAllBytes(Path.of("images/original.png")).toList()
        val expectedImage = Files.readAllBytes(Path.of("images/edit.png")).toList()
        assertEquals(
            actualImage,
            expectedImage,
            "The actual and the provided image are not identical"
        )
    }
}